tests/observability/metrics/utils.py

```diff
@@ -1251,3 +1261,42 @@ def validate_vnic_info(prometheus: Prometheus, vnic_info_to_compare: dict[str, s
         if actual_value != expected_value:
             mismatch_vnic_info[info] = {f"Expected: {expected_value}", f"Actual: {actual_value}"}
     assert not mismatch_vnic_info, f"There is a mismatch between expected and actual results:\n {mismatch_vnic_info}"
+
+
+def get_interface_name_from_vm(vm: VirtualMachineForTests):
+    interface_name = vm.privileged_vmi.virt_launcher_pod.execute(
+        command=shlex.split("bash -c \"virsh domiflist 1 | grep ethernet | awk '{print $1}'\"")
+    )
+    assert interface_name, f"Interface not found for vm {vm.name}"
+    return interface_name
+
+
+@contextmanager
+def create_windows10_vm(dv_name, namespace, client, vm_name, storage_class):
+    artifactory_secret = get_artifactory_secret(namespace=namespace)
+    artifactory_config_map = get_artifactory_config_map(namespace=namespace)
+    dv = DataVolume(
+        name=dv_name,
+        namespace=namespace,
+        storage_class=storage_class,
+        source="http",
+        url=get_http_image_url(image_directory=Images.Windows.UEFI_WIN_DIR, image_name=Images.Windows.WIN10_WSL2_IMG),
+        size=Images.Windows.DEFAULT_DV_SIZE,
+        client=client,
+        api_name="storage",
+        secret=artifactory_secret,
+        cert_configmap=artifactory_config_map.name,
+    )
+    dv.to_dict()
+    with VirtualMachineForTestsFromTemplate(
+        name=vm_name,
+        namespace=namespace,
+        client=client,
+        labels=Template.generate_template_labels(**py_config["latest_windows_os_dict"]["template_labels"]),
+        data_volume_template={"metadata": dv.res["metadata"], "spec": dv.res["spec"]},
+    ) as vm:
+        running_vm(vm=vm)
+        yield vm
```

Comment:
For all new tests, it would be better to create VMs with InstanceType, instead of Templates